hadoop-ozone/recon/src/main/java/org/apache/hadoop/ozone/recon/persistence/ContainerHealthSchemaManager.java

```diff
@@ -123,7 +126,26 @@ public void insertUnhealthyContainerRecords(List<UnhealthyContainers> recs) {
             rec.getContainerState());
       });
     }
-    unhealthyContainersDao.insert(recs);
+
+    try (Connection connection = containerSchemaDefinition.getDataSource().getConnection()) {
+      connection.setAutoCommit(false); // Turn off auto-commit for transactional control
+      for (UnhealthyContainers rec : recs) {
+        try {
+          unhealthyContainersDao.insert(rec);
+        } catch (DataAccessException dataAccessException) {
+          // Log the error and just update other fields of the existing record
+          // in case of ConstraintViolationException being actually thrown.
+          unhealthyContainersDao.update(rec);
+          LOG.error("Error while inserting unhealthy container record: {}", rec,
```

Comment:
Would it be a good idea to log both the old record and the new record that we will override?
Just a suggestion?

```
        // Fetch the existing record before updating
        UnhealthyContainers existingRecord = unhealthyContainersDao.get(rec.getContainerID());
        
        // Log both previous and new record details
        LOG.error("Duplicate entry detected for container ID {}. Updating existing record.\n"
                  + "Previous Record: {}\nNew Record: {}", 
                  rec.getContainerID(), existingRecord, rec);
        
        // Perform update
        unhealthyContainersDao.update(rec);
```

Reply:
I don't think, this will be helpful because ConstraintViolation comes when we get both containerId and containerState matches with same existing record, so if we log existing and current, it  will log same. So no difference.